indico/modules/events/contributions/operations.py

```diff
@@ -153,3 +155,30 @@ def delete_subcontribution(subcontrib):
     logger.info('Subcontribution %s deleted by %s', subcontrib, session.user)
     subcontrib.event_new.log(EventLogRealm.management, EventLogKind.negative, 'Subcontributions',
                              'Subcontribution "{}" has been deleted'.format(subcontrib.title), session.user)
+
+
+@no_autoflush
+def create_contribution_from_abstract(abstract, contrib_session=None):
+    event = abstract.event_new
+    contrib_person_links = {}
+    person_link_attrs = {'_title', 'address', 'affiliation', 'first_name', 'last_name', 'phone', 'author_type',
+                         'person', 'is_speaker'}
+    for abstract_person_link in abstract.person_links:
+        contrib_person_link_data = {}
+        for attr in person_link_attrs:
+            contrib_person_link_data[attr] = getattr(abstract_person_link, attr)
+        contrib_person_links[abstract_person_link.person.id] = ContributionPersonLink(**contrib_person_link_data)
```

Comment:
Why the dict by person ID? You cannot have the same person more than once in an abstract.